src/ucs/vfs/base/vfs_obj.c

```diff
@@ -371,6 +385,20 @@ static void ucs_vfs_path_list_dir_cb(ucs_vfs_node_t *node,
     }
 }
 
+/* must be called with lock held */
+static void ucs_vfs_node_build_unique_path(ucs_vfs_node_t *parent_node,
+                                           const char *name, char *path_buf,
+                                           size_t path_buf_size)
+{
+    size_t i = 1;
+
+    ucs_vfs_node_build_path(parent_node, name, path_buf, path_buf_size);
+    while (ucs_vfs_node_find_by_path(path_buf) != NULL) {
+        ucs_snprintf_safe(path_buf, path_buf_size, "%s_%zu", path_buf, i);
+        ++i;
+    }
```

Comment:
any way to avoid O(n^2) if we created N eps with same name?

Reply:
maybe use object pointer instead?